packages/eslint-plugin-nx/src/utils/runtime-lint-utils.ts

```diff
@@ -230,11 +246,7 @@ export function hasBannedImport(
       tags = [c.sourceTag];
     }
 
-    return (
-      c.bannedExternalImports &&
-      c.bannedExternalImports.length &&
-      tags.every((t) => (source.data.tags || []).includes(t))
-    );
+    return tags.every((t) => (source.data.tags || []).includes(t));
```

Comment:
Why was this removed? It's only relevant to `bannedExternalImports` and it saves a bit of computing time.

Reply:
Actually, this would break `allowedExternalImports` as it would filter out constraints that don't contain `bannedExternalImports` but contain `allowedExternalImports` and it would end up not calling `isConstraintBanningProject()`.

The real alternative would be `(c.bannedExternalImports && c.bannedExternalImports.length > 0) || c.allowedExternalImports && tags...` but it kind of duplicates the logic from `isConstraintBanningProject()` so we'd have to keep them in sync.
BTW, I don't recall well but I am pretty sure that at first I was surprised to notice that `isConstraintBanningProject()` wasn't called in the tests I added... until I noticed this condition. So this might surprises others in the future.

Is it really worth it in terms of performance?

If both `bannedExternalImports` & `allowedExternalImports` are null or undefined, then `isConstraintBanningProject()` doesn't have much to do except checking that they are both null or undefined.

What do you think?

Reply:
Yes, you are correct. Sorry for the confusion.